lms/services/http.py

```diff
@@ -0,0 +1,161 @@
+import requests
+from requests import RequestException
+
+from lms.services.exceptions import HTTPError, HTTPValidationError
+from lms.validation import ValidationError
+
+
+class HTTPService:
+    """Send HTTP requests with `requests` and receive the responses."""
+
+    def __init__(self, _session=None):
+        # A requests session is used so that cookies are persisted across
+        # requests and urllib3 connection pooling is used (which means that
+        # underlying TCP connections are re-used when making multiple requests
+        # to the same host, e.g. pagination).
+        #
+        # See https://docs.python-requests.org/en/latest/user/advanced/#session-objects
+        self._session = _session or requests.Session()
+
+    def request(
+        self,
+        method,
+        url,
+        params=None,
+        data=None,
+        json=None,
+        headers=None,
+        auth=None,
+        timeout=9,
+        schema=None,
+        **kwargs,
+    ):  # pylint:disable=too-many-arguments
+        """
+        Send a request with `requests` and return the requests.Response object.
```

Comment:
Note that no attempt is made to hide `requests` here: `HTTPService.request()` accepts arbitrary `kwargs` that are passed directly to `requests.Session.request()` and it also returns `requests.Response` objects directly. We _do_ wrap `requests`'s many different exception classes in our own two exception classes that have particular semantic meaning for us (`HTTPError` is for a problem with sending the request or receiving the successful response, `HTTPValidationError` is for a problem with validating the response).